### PR TITLE
Feature/cd rt sout net cdf

### PR DESCRIPTION
**Pull request by chengdang**
## Description
This PR add new module to write CRTM output RTS into netCDF format: 
`CRTM_RTSolution_WriteFile_netCDF`
https://github.com/JCSDA/CRTMv3/blob/6cd0398a56687f8371f1a0c320710d5e5aa1d8cf/src/RTSolution/CRTM_RTSolution_Define.f90#L1223

To use `CRTM_RTSolution_WriteFile_netCDF`, uncomment code block in `check_crtm.f90`:
https://github.com/JCSDA/CRTMv3/blob/6cd0398a56687f8371f1a0c320710d5e5aa1d8cf/src/Build/libsrc/test/check_crtm.F90#L393

There is no code changes to `src/RTSolution/Common_RTSolution.f90`, but a lot of auto space deletion due to code editor.

## Issue(s) addressed
https://github.com/JCSDA/CRTMv3/issues/62

## Dependencies
None

## Impact
None

### CODE REVIEW

File: src/RTSolution/CRTM_RTSolution_Define.f90 [hunk 1]
```diff
@@ -62,7 +67,7 @@ MODULE CRTM_RTSolution_Define
   PUBLIC :: CRTM_RTSolution_InquireFile
   PUBLIC :: CRTM_RTSolution_ReadFile
   PUBLIC :: CRTM_RTSolution_WriteFile
-
+  PUBLIC :: CRTM_RTSolution_WriteFile_netCDF
```
Comment (BenjaminTJohnson) [inline src/RTSolution/CRTM_RTSolution_Define.f90]: writefile_netcdf should live within CRTM_RTSolution_WriteFile -- you could pass an optional flag for it to determine if it's netCDF or not.    I would not at all be against just defaulting to netCDF as the standard output format and removing the binary option.

File: src/RTSolution/CRTM_RTSolution_Define.f90 [hunk 2]
```diff
@@ -110,6 +115,71 @@ MODULE CRTM_RTSolution_Define
   ! File status on close after write error
   CHARACTER(*), PARAMETER :: WRITE_ERROR_STATUS = 'DELETE'
 
+  ! Output netCDF attributes
```
Comment (BenjaminTJohnson) [inline src/RTSolution/CRTM_RTSolution_Define.f90:117]: How are these being used?

File: src/RTSolution/CRTM_RTSolution_Define.f90 [hunk 3]
```diff
@@ -134,7 +204,7 @@ MODULE CRTM_RTSolution_Define
     ! Forward radiative transfer intermediate results for a single channel
     !    These components are not defined when they are used as TL, AD
     !    and K variables
-    REAL(fp) :: SSA_Max                 = ZERO  ! Max Single Scattering Albedo in the profile                      
+    REAL(fp) :: SSA_Max                 = ZERO  ! Max Single Scattering Albedo in the profile
```
Comment (BenjaminTJohnson) [inline src/RTSolution/CRTM_RTSolution_Define.f90:210]: What is this used for?

File: src/RTSolution/CRTM_RTSolution_Define.f90 [hunk 4]
```diff
@@ -149,11 +219,11 @@ MODULE CRTM_RTSolution_Define
     REAL(fp), ALLOCATABLE :: Upwelling_Overcast_Radiance(:)   ! K
     REAL(fp), ALLOCATABLE :: Upwelling_Radiance(:)   ! K
     REAL(fp), ALLOCATABLE :: Layer_Optical_Depth(:)  ! K
-    REAL(fp), ALLOCATABLE :: Single_Scatter_Albedo(:)  ! K  
+    REAL(fp), ALLOCATABLE :: Single_Scatter_Albedo(:)  ! K
     ! Radiative transfer results for a single channel
     REAL(fp) :: Radiance               = ZERO
     REAL(fp) :: Brightness_Temperature = ZERO
-    REAL(fp) :: Stokes(4) 
+    REAL(fp) :: Stokes(4)
```
Comment (BenjaminTJohnson) [inline src/RTSolution/CRTM_RTSolution_Define.f90:229]: What are the contents of this?

**Review by BenjaminTJohnson — CHANGES_REQUESTED** [review]

File: src/RTSolution/CRTM_RTSolution_Define.f90 (hunk 1 above)
Comment (chengdang) [reply src/RTSolution/CRTM_RTSolution_Define.f90]: I can do that. Currently the `CRTM_RTSolution_WriteFile` is build as Binary output interface only. Let me try to modify it.

**Comment by chengdang** [comment]
Updated `CRTM_RTSolution_WriteFile` to write CRTM RTSolution object files in either Binary or NetCDF format.
The original `CRTM_RTSolution_WriteFile` has been modified as function `CRTM_RTSolution_WriteFile_Binary`. 
Users may continue use  `CRTM_RTSolution_WriteFile` for binary outputs, no changes to codes is needed.

**Comment by chengdang** [comment]
@BenjaminTJohnson This branch is ready for merge for netCDF output only. We should perhaps discuss more on the ctests related updates, see https://github.com/JCSDA/CRTMv3/issues/72.

All ctests passed but occasionally the channel subset test fails:
`test_k_matrix_ChannelSubset_iasi_metop-b `
I don't see any code change that may impact this mysterious test.

**Pull request merged**